ops/testing.py

```diff
@@ -380,8 +353,15 @@ def begin_with_initial_hooks(self) -> None:
             charm.on.leader_elected.emit()
         else:
             charm.on.leader_settings_changed.emit()
+
         charm.on.config_changed.emit()
+
         charm.on.start.emit()
+
+        # Set can_connect and fire pebble-ready for any containers.
+        for container_name in self._meta.containers:
+            self.set_can_connect(container_name, True)
```

Comment:
Here you are calling 'set_can_connect' for container in containers, but you *aren't* firing pebble_ready for container in containers.
Looking at the other code that is changing, it looks like "container_pebble_ready" fires the event and calls set_can_connect, but set_can_connect doesn't trigger pebble ready.
Maybe you wanted to call container_pebble_ready here?

Reply:
Ah yes, I documented it as calling pebble-ready, but then forgot to actually do it (and test it)! This is why we have code review ... will update shortly.